### PR TITLE
Bug 1519995 - Convert graphs view to React part 2

### PR DESCRIPTION
**Pull request by sarah-clements**
This converts most of the logic for the graphs view, connecting the UI controls, TestModal with the LegendCards and Graphs (all of the actual graph functionality like zoom, tooltip, etc is still in Angular - that'll be the next part). Some of this might still change so I'll write tests in the next pr, once angular has been removed completely from the graphs view.

I've launched this pr to [prototype](https://treeherder-prototype.herokuapp.com/perf.html#/graphs): @davehunt should this be tested by Ionut if he's back Monday or by bebe?  I think all of the functionality is working OK, with the caveat that if you are looking at a part of the graph that is zoomed, selecting the 'highlight alerts' toggle button will reset the zoom. This is a side affect of having the zoom fuctionality still in angular. If it's a deal breaker, I can hold off on merging this pr until all parts are done.

**Perf team**:
Let me know if you don't like how the graph has been constrained to a max width. I've done that with all the views, but if you aren't a fan I can either make it a bit wider or keep it as it was previously.

Something that I didn't do in this pull request that I'd like to do in the next is deprecate usage of the signature hash per bug [1519289](https://bugzilla.mozilla.org/show_bug.cgi?id=1519289). I think by switching to the signature id for the series param, we can also eliminate the need to store framework id and possibly repository in the query param. 

However, I realize that users might have certain graphs bookmarked so I propose a transition whereas if a signature hash is used in the query param or old style of `series=mozilla-central,12345,1` is used, the UI will update the params to how they will be going forward and an alert can pop up mentioning that the old style/signature hash has been deprecated and to please update bookmarks. We can also discuss this at the next Perfherder planning meeting if you prefer.

FYI @karlht

### CODE REVIEW

**Comment by davehunt** [comment]
Thanks @sarah-clements I think it's looking great. I'll let @ionutgoldan and @bebef1987 comment on the graph width and zoom reset, as they're likely to be the most impacted as perf sheriffs. If we can hold off merging until @ionutgoldan has a chance to review that would be great!

**Comment by sarah-clements** [comment]
> Thanks @sarah-clements I think it's looking great. I'll let @ionutgoldan and @bebef1987 comment on the graph width and zoom reset, as they're likely to be the most impacted as perf sheriffs. If we can hold off merging until @ionutgoldan has a chance to review that would be great!

Sure, I can definitely wait for Ionut to review before merging.

File: tests/selenium/pages/perfherder.py [hunk 1]
```diff
@@ -6,15 +6,16 @@
                    Modal)
 
 
+# TODO fix tests to work with react
```
Comment (camd) [inline tests/selenium/pages/perfherder.py]: I presume this means it'll come in a follow-up PR.  :)

File: ui/perfherder/graphs/GraphsView.jsx [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    this.setState({ loading: true });
+
+    const responses = await Promise.all(
+      tests.map(series =>
+        getData(
+          createApiUrl(perfSummaryEndpoint, this.createSeriesParams(series)),
+        ),
+      ),
+    );
+    const errorMessages = processErrors(responses);
```
Comment (camd) [inline ui/perfherder/graphs/GraphsView.jsx]: I think this is the pattern you referenced in your comments on my other PR.  I'll adjust to using this pattern.  It looks tidy.  :)

Comment (sarah-clements) [reply ui/perfherder/graphs/GraphsView.jsx]: Well there's two - `processResponse` which is for more general use. This `processErrors` is for a specific use case of dealing with only errors in the array returned when using `Promise.all` (and I either don't care about the response if the update was OK or I need to munge the data before setting state).

File: tests/selenium/pages/perfherder.py (hunk 1 above)
Comment (sarah-clements) [reply tests/selenium/pages/perfherder.py]: Yes, that'll be in the next pr. I'm also Ok with holding off on merging until the final part is converted and tests added (or fixed).

**Comment by ionutgoldan** [comment]
> **Perf team**:
> Let me know if you don't like how the graph has been constrained to a max width. I've done that with all the views, but if you aren't a fan I can either make it a bit wider or keep it as it was previously.

It would better to keep the graph's previous width. It's easier on the eyes when investigating.

**Comment by sarah-clements** [comment]
> > **Perf team**:
> > Let me know if you don't like how the graph has been constrained to a max width. I've done that with all the views, but if you aren't a fan I can either make it a bit wider or keep it as it was previously.
> 
> It would better to keep the graph's previous width. It's easier on the eyes when investigating.

No problem, I can keep leave it at the previous width.

**Comment by sarah-clements** [comment]
Going to squash this and rebase my final graphs pr onto it.

**Pull request closed without merging**